packages/@uppy/provider-views/src/ProviderView/ProviderView.tsx

```diff
@@ -328,6 +565,37 @@ export default class ProviderView<M extends Meta, B extends Body> {
     return result
   }
 
+  /**
+   * We still build the ancestor path when the user checks or unchecks a search result.
+   * Building ancestor nodes isn’t expensive anymore since it doesn’t involve network calls.
+   * Even if a checked item is later unchecked without being uploaded, it still gets added to the partialTree.
+   *
+   * While it might seem intuitive to build the ancestor path only when opening a folder and,
+   * when checking and uploading a file/folder from search view, that approach would require patching edge cases
+   * related to checked state across the two views ( Search View and Normal View) in openFolder and afterOpenFolder.
```

Comment:
I found your comment you placed above here with the edge much more revealing on why this is needed than this